client/src/pages/Home/DebugBar.tsx

```diff
@@ -44,90 +47,53 @@ function DebugBar({
   const debugImgId = "sampleStoreQRImgSrc";
   const uploadImgId = "storeQRImgSrc";
 
-  // Dummy map attachment
-  const map = new google.maps.Map(document.getElementById("mapPlaceholder"));
-  // Declare our Google Map/Places API service handler
-  const service = new google.maps.places.PlacesService(map);
-
-  // Error callback for location API call
-  const errorbackLoc = (error: any) => {
-    alert(error);
-  };
-
   // Grab the location of device
   const grabLoc = () => {
-    // Use browser navigator if we are in web environment & available
-    if (isWeb) {
-      if (navigator.geolocation) {
-        setIsLoading(true);
-        navigator.geolocation.getCurrentPosition(fetchStores, errorbackLoc);
-      }
-    } else {
-      setIsLoading(true);
-      microapps
-        .getCurrentLocation()
-        .then((loc: any) => {
-          const position = {
-            coords: {
-              latitude: loc["latitude"],
-              longitude: loc["longitude"],
-            },
-          };
-          fetchStores(position);
-        })
-        .catch((err: any) => errorbackLoc(err));
-    }
+    setIsLoading(true);
+    getGeoLocation(geoLocationCallback);
   };
 
-  // fetch list of users
-  const fetchStores = async (position: {
-    coords: {
-      latitude: number;
-      longitude: number;
-    };
-  }) => {
-    // Update location
-    const data = {
-      distance: 10000,
-      latitude: position.coords.latitude,
-      longitude: position.coords.longitude,
-    };
-    const stores = await fetchJson("POST", data, STORE_LIST_API);
-    setUserCoords([position.coords.latitude, position.coords.longitude]);
-    storesCallback(stores);
+  const geoLocationCallback = (position: GeoLocation) => {
     setIsLoading(false);
+    setCurGeoLocation(position);
+  };
+
+  const fetchStores = async (position?: GeoLocation) => {
+    // When called without arguments, we default to using
+    // already present location previously retrieved
+    if (!position) {
+      if (curGeoLocation) {
+        position = curGeoLocation;
+      } else {
+        return;
```

Comment:
is this a return condition or error? we should always have location in one way or another right?

Reply:
I return nothing here since we cannot yet fetch with alphabetical ordering (at least till after the next PR)